.github/workflows/sync-with-upstream.yaml

```diff
@@ -0,0 +1,78 @@
+# Create a PR to sync with the upstream template repo.
+# The template repo is https://github.com/jenstroeger/python-package-template.
+
+name: Sync with template repository
+on:
+  schedule:
+  - cron: 11 11 * * *
+
+permissions:
+  contents: read
+env:
+  # This is the username and email for the user who creates a branch and commits
+  # the changes. In an organisation that should be a dedicated devops account.
+  USER_NAME: jenstroeger
+  USER_EMAIL: jenstroeger@users.noreply.github.com
+
+jobs:
+  sync:
+    if: github.repository != 'jenstroeger/python-package-template'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+
+    - name: Check out template repository
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      with:
+        repository: jenstroeger/python-package-template
```

Comment:
Need the `token: ${{ secrets.... }}` here to make this work, you can't checkout without the token

Reply:
> Need the `token: ${{ secrets.... }}` here to make this work, you can't checkout without the token

Have you tried this workflow with this [template repo](https://github.com/jenstroeger/python-package-template)? The token should not be needed for checking out this template repo because it's public.

Reply:
I think the `token:` is required if the upstream template repository is private. Maybe we should leave a comment about that in this workflow?

Reply:
> I think the `token:` is required if the upstream template repository is private. Maybe we should leave a comment about that in this workflow?

Commit [3c4a557](https://github.com/jenstroeger/python-package-template/pull/444/commits/3c4a5570df074d3d4e99efaa0aee1a22f7b87628).